pkg/proxy/keepalive/keepalive.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keepalive
+
+import (
+	"net"
+
+	"github.com/pingcap/TiProxy/lib/config"
+	"github.com/pingcap/TiProxy/lib/util/errors"
+)
+
+var (
+	ErrKeepAlive = errors.New("failed to set keepalive")
+)
+
+func SetKeepalive(conn net.Conn, cfg config.KeepAlive) error {
+	if !cfg.Enabled {
```

Comment:
It should be enabled by default; if it's false, call setKeepalive(false).

Reply:
Is it possible that it's true by default? So if it's false, you need to disable it explicitly.

Reply:
What about now?

Reply:
- It's a bad idea to name a configuration item `disabled`, because it's inconsistent with other configurations. Previous TiDB users complained about this.
- It doesn't solve the problem: the keep alive may be enabled and then disabled multiple times, since the configuration is supposed to be dynamic. So you need to set it explicitly every time it changes.

Reply:
There is no way that it can be enabled by default. There are total 4 syscall options. I split them into `SetKeepalive` and `SetKeepaliveParams`. See if that suits you.